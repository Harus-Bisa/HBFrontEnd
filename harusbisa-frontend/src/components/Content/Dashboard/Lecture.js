import React from "react";
import {connect} from "react-redux";
import QuizCard from "../../Card/QuizCard";
import { Button } from "@material-ui/core";
import Popup from "../../Popup/Popup";

function mapStateToProps(state){
    return {
        lecture: state.selectedLecture
    }
}
function Lecture(props){
    const makeQuizzes = () =>{
        var quizzes = []
        for (let i=0; i<props.lecture.quizzes.length; i++){
            quizzes.push(<QuizCard key={i} index={i}/>)
        }
        return quizzes;
    }
    if (!props.lecture){
        return(
            <div>Make your first lecture</div>
        )
    }
    return(
        <div className="content">
            <header>
                <h1>Sesi {props.lecture.date}</h1>
            </header>
            <div style={{margin:"1rem 0"}}>
                <div className="row">
                    <div className="col-3">
                        <Button fullWidth className="lecture-button">Buka kelas</Button>
                    </div>
                    <div className="col-3">
                        <Popup
                            purpose="Tambah Pertanyaan"
                            trigger={{component:Button, className:"lecture-button"}}
                        />
                    </div>
                    <div className="col-3">
                        <Popup
                            purpose= {"Statistik sesi"+ props.lecture.date}
                            trigger={{component:Button, className:"lecture-button"}}
                        />
                    </div>
                    <div className="col-3">
                        <Popup
                            purpose= {"Setting sesi"+props.lecture.date}
                            trigger={{component:Button, className:"lecture-button"}}
                        />
                    </div>
                </div>
            </div>
            <div>
                {makeQuizzes()}
            </div>
        </div>
    )
}

export default connect(mapStateToProps)(Lecture);
