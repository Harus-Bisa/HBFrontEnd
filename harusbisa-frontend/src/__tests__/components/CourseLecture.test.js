import React from "react";
import {shallow, mount} from "enzyme";
import configureMockStore from "redux-mock-store";
import '../../test-config';
import {Provider} from 'react-redux';
import Lecture from "../../components/Content/Dashboard/Lecture";
const mockStore = configureMockStore();

describe("Course Lecture component", () =>{
    it("matches snapshot", () =>{
        let initialState = {
            selectedLecture: {
                date: 1574897055773,
                quizzes:[{
                    "question": "1 + 1 = ?",
                    "answerOptions": ["4","2","0"]
                }]
            },
            course:{
                courseName:'abc'
            }
        }
        let store = mockStore(initialState);
        let component = shallow(<Provider store={store}><Lecture/></Provider>)
        expect(component.html()).toMatchSnapshot()
    })
})