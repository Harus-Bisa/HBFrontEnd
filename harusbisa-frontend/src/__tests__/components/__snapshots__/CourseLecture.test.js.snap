// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Lecture component matches snapshot 1`] = `"<div class=\\"content\\"><header><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">abc</span></button><h1>Sesi 27/11/2019</h1></header><div style=\\"margin:1rem 0\\"><div class=\\"row\\"><div class=\\"col-3\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text lecture-button MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><div><div class=\\"row\\"><div class=\\"col-4\\" style=\\"display:flex\\"><svg class=\\"MuiSvgIcon-root student-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\" style=\\"margin:auto\\"><path d=\\"M8 5v14l11-7z\\"></path></svg></div><div class=\\"col-8\\" style=\\"display:flex\\"><p style=\\"margin:auto\\">Buka kelas</p></div></div></div></span></button></div><div class=\\"col-3\\"><div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text lecture-button\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><div><div class=\\"row\\"><div class=\\"col-4\\" style=\\"display:flex\\"><svg class=\\"MuiSvgIcon-root student-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\" style=\\"margin:auto\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></div><div class=\\"col-8\\" style=\\"display:flex\\"><p style=\\"margin:auto\\">Tambah Pertanyaan</p></div></div></div></span></button></div></div><div class=\\"col-3\\"><div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text lecture-button\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><div><div class=\\"row\\"><div class=\\"col-4\\" style=\\"display:flex\\"><svg class=\\"MuiSvgIcon-root student-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\" style=\\"margin:auto\\"><path d=\\"M10 20h4V4h-4v16zm-6 0h4v-8H4v8zM16 9v11h4V9h-4z\\"></path></svg></div><div class=\\"col-8\\" style=\\"display:flex\\"><p style=\\"margin:auto\\">Statistik Sesi 27/11/2019</p></div></div></div></span></button></div></div><div class=\\"col-3\\"><div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text lecture-button\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><div><div class=\\"row\\"><div class=\\"col-4\\" style=\\"display:flex\\"><svg class=\\"MuiSvgIcon-root student-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\" style=\\"margin:auto\\"><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg></div><div class=\\"col-8\\" style=\\"display:flex\\"><p style=\\"margin:auto\\">Setting Sesi 27/11/2019</p></div></div></div></span></button></div></div></div></div><div><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root card MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><div class=\\"row\\"><div class=\\"col-1\\"><p>1</p></div><div class=\\"col-9\\"><p>1 + 1 = ?</p></div><div class=\\"col-2\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg></span></button></div></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height:0px\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div><div><p>A. 4</p></div><div><p>B. 2</p></div><div><p>C. 0</p></div></div></div></div></div><div class=\\"MuiCardActions-root MuiCardActions-spacing\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path></svg></span></button></div></div></div></div>"`;
